src/PSM3.sol

```diff
@@ -208,6 +207,7 @@ contract PSM3 is IPSM3, Ownable {
             : maxAssetsToWithdraw;
 
         // Get shares to burn, rounding up for both calculations
+        // NOTE: Don't need to check valid asset here since `_getAssetValue` will revert if invalid
```

Comment:
You didn't remove the `_isValidAsset` on line 201.

Reply:
This actually reverts on `asset.balanceOf(address(this))` with an EVM revert for a non-erc20 compliant asset so opted to leave this check in withdraw, forgot to remove comment though, just removed

Reply:
ok makes sense